### ISSUE TITLE
feat: Wake word detection ("Hey OpenHush") with circular pre-buffer

### ISSUE THREAD

**Issue by claymore666** [claymore666/openhush#63]
## Summary

Enable hands-free activation via customizable wake word (like "Hey Alexa", "OK Google", "Hey Siri") using a circular audio pre-buffer that's always listening.

## Motivation

Currently OpenHush requires a hotkey to activate. A wake word would enable:
- **Hands-free operation** - No keyboard needed
- **Accessibility** - Users who can't use keyboard
- **Natural interaction** - Just start talking
- **Smart home style** - "Hey OpenHush, type an email to John"

## Architecture

```
┌─────────────────────────────────────────────────────────────────┐
│                    Circular Pre-Buffer                          │
│                  (always recording last 30s)                    │
└─────────────────────────────────────────────────────────────────┘
                              │
                              ▼
              ┌───────────────────────────────┐
              │   Wake Word Detector (CPU)    │
              │   - Lightweight model         │
              │   - Runs continuously         │
              │   - Low latency (<100ms)      │
              └───────────────────────────────┘
                              │
                    [wake word detected]
                              │
                              ▼
              ┌───────────────────────────────┐
              │   Command Capture Mode        │
              │   - Record until silence      │
              │   - Or until "stop" word      │
              │   - Send to GPU for STT       │
              └───────────────────────────────┘
                              │
                              ▼
              ┌───────────────────────────────┐
              │   Transcribe & Output         │
              └───────────────────────────────┘
```

### 1. Circular Pre-Buffer (Prerequisite)

Always-on recording that solves multiple issues:
- **#57**: No audio delay on hotkey press
- **#62**: Decoupled recording architecture
- **This**: Wake word detection source

```toml
[audio]
pre_buffer_secs = 30        # Keep last 30 seconds
pre_buffer_enabled = true   # Can disable to save resources
```

### 2. Wake Word Detection

```toml
[wake_word]
enabled = true
phrase = "hey open hush"    # Customizable
sensitivity = 0.5           # 0.0 (strict) to 1.0 (loose)
timeout_secs = 10           # Max command length after wake
end_phrase = "stop"         # Optional stop word
```

**Custom wake words:**
- "Hey OpenHush"
- "Computer" (Star Trek style)
- "Dictate"
- User's name
- Any custom phrase

#### Option A: Keyword Spotting Model (Recommended)
- Use lightweight models like **OpenWakeWord**, **Porcupine**, or **Snowboy**
- Runs on CPU, ~1% usage
- Pre-trained for common patterns, fine-tunable

```rust
// Using openwakeword or similar
let detector = WakeWordDetector::new("hey open hush")?;

loop {
    let audio_chunk = pre_buffer.get_latest(100ms);
    if detector.detect(&audio_chunk) {
        start_command_capture();
    }
}
```

#### Option B: Whisper-based (Simpler but heavier)
- Run Whisper `tiny` model on small chunks
- Check if transcription contains wake phrase
- Higher resource usage but no extra dependency

#### Option C: Simple Energy + Pattern (Lightest)
- Voice activity detection first
- Then check audio fingerprint against wake phrase template
- Lowest accuracy but zero dependencies

### 4. Command Modes

After wake word detection:

**Mode 1: Silence-terminated**
```
"Hey OpenHush, send an email to John about the meeting tomorrow" [pause]
→ Transcribes everything after wake word until silence
```

**Mode 2: Stop-word terminated**
```
"Hey OpenHush, write a long paragraph about... [content] ...stop"
→ Transcribes until "stop" word detected
```

**Mode 3: Voice command**
```
"Hey OpenHush, start dictation"
→ Enters continuous dictation mode (like holding hotkey)
"Hey OpenHush, stop dictation"
→ Exits dictation mode
```

## Privacy Considerations

⚠️ **Always-on microphone has privacy implications**

Mitigations:
1. **Local only** - All processing on-device, no cloud
2. **Visual indicator** - Tray icon shows listening state
3. **Audio indicator** - Optional beep when wake word detected
4. **Easy disable** - Quick toggle in tray menu
5. **No storage** - Pre-buffer is circular, old audio discarded
6. **Mute support** - Hardware mute button respected

```toml
[wake_word]
enabled = true
indicator_visual = true     # Tray icon changes when listening
indicator_audio = true      # Beep on wake word detection
respect_hw_mute = true      # Honor hardware mute switch
```

### Phase 1: Circular Pre-Buffer
- [ ] Implement ring buffer for audio samples
- [ ] Integrate with existing audio capture
- [ ] Fix #57 (hotkey delay) as side effect
- [ ] Add config options

### Phase 2: Wake Word Detection
- [ ] Integrate OpenWakeWord or similar library
- [ ] Add wake phrase configuration
- [ ] Implement detection loop (separate thread)
- [ ] Add sensitivity tuning

### Phase 3: Command Capture
- [ ] Silence detection for command end
- [ ] Optional stop word detection
- [ ] Integration with existing transcription pipeline

### Phase 4: Voice Commands
- [ ] "Start/stop dictation" commands
- [ ] "Cancel" command
- [ ] Custom command phrases

## Dependencies

Potential wake word libraries:
- [OpenWakeWord](https://github.com/dscripka/openWakeWord) - Python, accurate, customizable
- [Porcupine](https://github.com/Picovoice/porcupine) - Cross-platform, free tier
- [Rustpotter](https://github.com/GiviMAD/rustpotter) - Rust native! 
- [Vosk](https://alphacephei.com/vosk/) - Already used by nerd-dictation

**Rustpotter** looks promising - Rust native wake word detection!

## Related Issues

- #57 - Audio recording delay (pre-buffer fixes this)
- #62 - Decoupled recording architecture (pre-buffer prerequisite)
- #61 - System audio capture (different source, same buffer concept)

## Competitor Analysis

| Tool | Wake Word |
|------|-----------|
| Alexa/Siri/Google | ✅ Cloud-based |
| Dragon | ✅ "Wake up" |
| Talon | 🟡 Noise-based (pop sound) |
| Others | ❌ None |

This would make OpenHush unique among open-source dictation tools.
